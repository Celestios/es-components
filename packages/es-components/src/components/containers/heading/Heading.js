import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const HeadingBase = styled.h1`
  color: inherit;
  font-size: ${props => props.theme.headingSize[props.adjustedSize]};
  font-weight: 300;
  line-height: 1.1;
  margin-bottom: 12.5px;
  margin-top: 25px;

  small {
    font-size: ${props => (props.adjustedSize > 3 ? '75%' : '65%')};
    line-height: 1;
  }
`;

function Heading({ children, level, size, ...other }) {
  const adjustedSize = size || level;
  const hLevel = `h${level}`;
  return (
    <HeadingBase as={hLevel} adjustedSize={adjustedSize} {...other}>
      {children}
    </HeadingBase>
  );
}

Heading.propTypes = {
  children: PropTypes.node,
  /** Heading level element */
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  /** Override the default font size with another level */
  size: PropTypes.oneOf([1, 2, 3, 4, 5, 6])
};

Heading.defaultProps = {
  children: undefined,
  size: undefined
};

export default Heading;
