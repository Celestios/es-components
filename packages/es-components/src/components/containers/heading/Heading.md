A `Heading` component can be used for various heading levels. Wrap secondary text in `<small>` tags. Provide an `underlineColor` to display a styled border.
Larger headings will resize appropriately in mobile viewports.

```
const StyleReset = require('../../util/StyleReset').default;

<div>
  <StyleReset />
  <Heading level={1} underlineColor="#00a0d2">h1. Heading <small>Secondary Text</small></Heading>
  <Heading level={2} underlineColor="#979797">h2. Heading <small>Secondary Text</small></Heading>
  <Heading level={3}>h3. Heading <small>Secondary Text</small></Heading>
  <Heading level={4}>h4. Heading <small>Secondary Text</small></Heading>
  <Heading level={5}>h5. Heading <small>Secondary Text</small></Heading>
  <Heading level={6}>h6. Heading <small>Secondary Text</small></Heading>
</div>
```

Use `isKnockoutStyle` to display a special page-level heading style.

```
<Heading level={1} isKnockoutStyle>Knockout Heading</Heading>
```

The font size may be overridden by that of another level by providing the `size` prop.

```
<div>
  <Heading level={1} size={3}>h1. Heading in h3 size</Heading>
</div>
```
