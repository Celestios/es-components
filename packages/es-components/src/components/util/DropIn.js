import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';

const DropIn = ({ children, duration, ...otherProps }) => {
  const transitionStyles = {
    opacity: 0,
    transform: `translate(0, -25%)`,
    transition: `transform ${duration}ms ease-out, opacity ${duration}ms linear`
  };

  const stateStyles = {
    entering: { opacity: 1, transform: `translate(0, 0)` },
    entered: { opacity: 1, transform: `translate(0, 0)` }
  };

  return (
    <Transition {...otherProps} timeout={duration}>
      {(state, innerProps) =>
        React.cloneElement(children, {
          ...innerProps,
          style: {
            ...children.props.style,
            ...transitionStyles,
            ...stateStyles[state]
          }
        })
      }
    </Transition>
  );
};

DropIn.propTypes = {
  children: PropTypes.any.isRequired,
  duration: PropTypes.number
};

DropIn.defaultProps = {
  duration: 300
};

export default DropIn;
