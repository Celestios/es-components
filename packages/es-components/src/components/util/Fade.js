import React from 'react';
import PropTypes from 'prop-types';
import Transition from 'react-transition-group/Transition';

const Fade = ({ children, duration, opacity, ...otherProps }) => {
  const transitionStyles = {
    transition: `opacity ${duration}ms linear`,
    opacity: 0
  };

  const stateStyles = {
    entering: { opacity },
    entered: { opacity }
  };

  return (
    <Transition {...otherProps} timeout={duration}>
      {(state, innerProps) =>
        React.cloneElement(children, {
          ...innerProps,
          style: {
            ...children.props.style,
            ...transitionStyles,
            ...stateStyles[state]
          }
        })
      }
    </Transition>
  );
};

Fade.propTypes = {
  children: PropTypes.any.isRequired,
  duration: PropTypes.number,
  opacity: PropTypes.number
};

Fade.defaultProps = {
  duration: 150,
  opacity: 1
};

export default Fade;
