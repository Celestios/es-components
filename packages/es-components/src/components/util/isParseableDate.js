import parse from 'date-fns/parse';
import isValid from 'date-fns/isValid';

export default function isParseableDate(date) {
  const parsedDate = parse(date);
  return isValid(parsedDate);
}
