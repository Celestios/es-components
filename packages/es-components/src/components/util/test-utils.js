// eslint-env jest
import React from 'react';
import { ThemeProvider } from 'styled-components';
import viaTheme from 'es-components-via-theme';
import { render } from '@testing-library/react';

export function renderWithTheme(component) {
  return render(
    <ThemeProvider theme={viaTheme}>
      {React.cloneElement(component)}
    </ThemeProvider>
  );
}
