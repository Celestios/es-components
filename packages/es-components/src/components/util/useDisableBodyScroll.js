import { useEffect } from 'react';

export const useDisableBodyScroll = shouldDisableScroll => {
  useEffect(() => {
    const newStyle = 'overflow: hidden;';
    const getCurrentStyle = () => document.body.getAttribute('style') || '';
    const disableScroll = () =>
      document.body.setAttribute('style', `${getCurrentStyle()} ${newStyle}`);
    const enableScroll = () =>
      document.body.setAttribute(
        'style',
        getCurrentStyle().replace(newStyle, '')
      );

    if (shouldDisableScroll) {
      disableScroll();
    } else {
      enableScroll();
    }

    return enableScroll;
  }, [shouldDisableScroll]);
};
