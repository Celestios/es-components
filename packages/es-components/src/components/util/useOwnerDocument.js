import { useState, useCallback } from 'react';

export default function useOwnerDocument(currentDocument) {
  const [ownerDocument, setOwnerDocument] = useState(currentDocument);
  const ownerDocumentRef = useCallback(node => {
    if (!node) return;
    const targetDocument =
      node.ownerDocument?.body || node.ownerDocument || currentDocument;
    if (ownerDocument !== targetDocument) setOwnerDocument(targetDocument);
  });
  return [ownerDocument, ownerDocumentRef];
}
