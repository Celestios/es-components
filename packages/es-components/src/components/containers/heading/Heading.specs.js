/* eslint-env jest */

import React from 'react';
import { renderWithTheme } from '../../util/test-utils';

import Heading from './Heading';

it('renders proper heading level', () => {
  const { queryByText } = renderWithTheme(
    <Heading level={1}>Heading level 1</Heading>
  );

  expect(queryByText('Heading level 1')).not.toBeNull();
});

it('renders heading level with another size', () => {
  const { container } = renderWithTheme(
    <div>
      <Heading level={1}>Heading level 1</Heading>
      <Heading level={3}>Heading level 3</Heading>
      <Heading level={3} size={1}>
        Heading level 3 size 1
      </Heading>
    </div>
  );

  expect(container).toMatchSnapshot();
});

it('renders knockout heading with different class', () => {
  const { container } = renderWithTheme(
    <div>
      <Heading level={1}>Heading level 1</Heading>
      <Heading level={1} isKnockoutStyle>
        Knockout Heading level 1
      </Heading>
    </div>
  );

  expect(container).toMatchSnapshot();
});
