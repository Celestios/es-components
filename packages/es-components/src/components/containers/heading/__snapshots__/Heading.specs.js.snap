// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heading level with another size 1`] = `
<div>
  <div>
    <h1
      class="sc-bdVaJa gbxEJg"
    >
      Heading level 1
    </h1>
    <h3
      class="sc-bdVaJa bbqOIv"
    >
      Heading level 3
    </h3>
    <h3
      class="sc-bdVaJa gbxEJg"
    >
      Heading level 3 size 1
    </h3>
  </div>
</div>
`;
